Mono: Implement stack info for errors and exceptions

neikeq opened the pull request:
Closes #14589

![ss1](https://user-images.githubusercontent.com/7718100/34654371-0ce69706-f3fb-11e7-8a4c-2d8513fe5541.png)

![ss2](https://user-images.githubusercontent.com/7718100/34654373-0dfd3244-f3fb-11e7-9e7e-1688898406c9.png)

neikeq commented on core/script_debugger_remote.cpp:
```diff
@@ -899,6 +863,41 @@ void ScriptDebuggerRemote::send_message(const String &p_message, const Array &p_
 	mutex->unlock();
 }
 
+void ScriptDebuggerRemote::send_error(const String &p_func, const String &p_file, int p_line, const String &p_err, const String &p_descr, ErrorHandlerType p_type, const Vector<ScriptLanguage::StackInfo> &p_stack_info) {
```
@reduz As I mentioned on IRC, I needed to add this function in order to display the exception stack info.

akien-mga commented:
Your branch is *old*, dude ;) https://github.com/neikeq/godot/commits/the-stack-frame-madness

neikeq committed:
Mono: Implement stack info for errors and exceptions

neikeq committed:
External editor fixes
- Fix VS Code opening on the previous line to the desired one.
- Fix running MonoDevelop without the line and column parameters.
- Fix `ScriptEditor::_goto_script_line` not working with language overriden external editors.

neikeq commented:
That is discrimination against old branches.

neikeq commented on core/script_language.h:
```diff
@@ -254,7 +254,8 @@ class ScriptLanguage {
-		Ref<Script> script;
+		String file;
+		String func;
 		int line;
```
Forgot to mention I changed the StackInfo struct. It includes the file path and the function string (not necessarily the name, it can be formatted to include the signature). I also removed the script from it because it was not needing and it meant you couldn't include files outside the project.

neikeq commented on editor/script_editor_debugger.cpp:
```diff
@@ -1616,30 +1616,33 @@ void ScriptEditorDebugger::_error_selected(int p_idx) {
+		md.push_back(st[i + 2]);
 
-		String str = script.get_file() + ":" + itos(line);
+		String str = func + " in " + script.get_file() + ":line " + itos(line);
```
Also a slightly different formatting for the stack frame.
